### PR TITLE
wrong shortcode

### PR DESCRIPTION
**Pull request by kavehmz**

### CODE REVIEW

File: lib/Finance/Contract.pm [hunk 1]
```diff
@@ -754,27 +765,23 @@ sub _build_date_start {
...
-
-    $string /= _FOREX_BARRIER_MULTIPLIER if $contract_type !~ /^DIGIT/ and $string and looks_like_number($string);
-
-    return $string;
-}
-
 # Generates a string version of a barrier by multiplying the actual barrier to remove the decimal point
 sub _barrier_for_shortcode_string {
-    my ($self, $string, $contract_type) = @_;
+    my ($self, $string) = @_;
```
Comment (chylli) [inline lib/Finance/Contract.pm]: write a test please ?

File: lib/Finance/Contract.pm [hunk 2]
```diff
@@ -754,27 +765,23 @@ sub _build_date_start {
     return Date::Utility->new;
 }
 
-# Generates a barrier value from the string used in a shortcode
-sub _barrier_from_shortcode_string {
```
Comment (chylli) [inline lib/Finance/Contract.pm]: Isn't it used anywhere yet ? Needn't change other place that call this function ?

File: lib/Finance/Contract.pm [hunk 3]
```diff
@@ -754,27 +765,23 @@ sub _build_date_start {
...
 
     # Do not manipulate relative barriers.
     return $string if not looks_like_number($string);
 
     $string = $self->_pipsized_value($string);
-    $string *= _FOREX_BARRIER_MULTIPLIER if $contract_type !~ /^DIGIT/ and $string and looks_like_number($string);
+    if ($self->absolute_barrier_multiplier) {
+        $string *= _FOREX_BARRIER_MULTIPLIER;
+    } else {
+        $string = floor($string);
```
Comment (raunakkathuria) [inline lib/Finance/Contract.pm]: why we are using floor? we were not using it before

Comment (kavehmz) [inline lib/Finance/Contract.pm]: It was in the main logic. For now I prefer to move the logic intact as there was enough bugs in the logic change up to now.

**Pull request merged**